server/tserver/src/main/java/org/apache/accumulo/tserver/replication/AccumuloReplicaSystem.java

```diff
@@ -416,6 +444,12 @@ protected Status replicateLogs(ClientContext peerContext, final HostAndPort peer
     Status lastStatus = status, currentStatus = status;
     final AtomicReference<Exception> exceptionRef = new AtomicReference<>();
     while (true) {
+      // Check if the thread has been interrupted prior to replicating data
+      if (Thread.interrupted()) {
```

Comment:
In the past I have found that other libraries (like Hadoop) eat interrupts, so it may never be seen.  I am not sure, but I was wondering if it would it make sense to have an additional atomic boolean that you set and and check?  I had to do this to reliably interrupt scans in Accumulo. If there were an AtomicBoolean, could do something like `if(Thread.interrupted() || stopped.get())`.